cumulusci/core/keychain/encrypted_file_project_keychain.py

```diff
@@ -160,12 +89,410 @@ def unset_default_org(self):
+        """Loads .org files in a given directory onto the keychain"""
+        if not dirname:
+            return
+        dir_path = Path(dirname)
+        for item in sorted(dir_path.iterdir()):
+            if item.suffix == ".org":
+                with open(item, "r") as f:
+                    config = f.read()
+                name = item.name.replace(".org", "")
+                if "orgs" not in self.config:
+                    self.config["orgs"] = {}
+                self.config["orgs"][name] = (
+                    constructor(config) if constructor else config
+                )
+
+    def _set_encrypted_org(self, name, encrypted, global_org):
+        if global_org:
+            filename = Path(f"{self.global_config_dir}/{name}.org")
+        elif self.project_local_dir is None:
+            return
+        else:
+            filename = Path(f"{self.project_local_dir}/{name}.org")
+        with open(filename, "wb") as f_org:
+            f_org.write(encrypted)
+
+    def _get_org(self, name):
+        org = self._decrypt_config(
+            OrgConfig,
+            self.orgs[name].encrypted_data,
+            extra=[name, self],
+            context=f"org config ({name})",
+        )
+        if self.orgs[name].global_org:
+            org.global_org = True
+        return org
+
+    def _remove_org(self, name, global_org):
+        scope = self.global_config_dir if global_org else self.project_local_dir
+        org_path = Path(f"{scope}/{name}.org")
+        if not org_path.exists():
+            if not global_org:
+                raise OrgNotFound(
+                    f"Could not find org named {name} to delete.  Deleting in project org mode.  Is {name} a global org?"
+                )
+            raise OrgNotFound(
+                f"Could not find org named {name} to delete.  Deleting in global org mode.  Is {name} a project org instead of a global org?"
+            )
+
+        org_path.unlink()
+        del self.orgs[name]
+
+    def _raise_org_not_found(self, name):
+        raise OrgNotFound(
+            f"Org information could not be found. Expected to find encrypted file at {self.project_local_dir}/{name}.org"
+        )
+
+    #######################################
+    #              Services               #
+    #######################################
+
+    def set_default_service(
+        self, service_type: str, alias: str, project: bool = False
+    ) -> None:
+        """Public API for setting a default service e.g. `cci service default`"""
+        if service_type not in self.project_config.services:
+            raise ServiceNotConfigured(f"No such service type: {service_type}")
+        elif alias not in self.services[service_type]:
+            raise ServiceNotConfigured(
+                f"No service of type {service_type} configured with name: {alias}"
+            )
+        elif alias == self._default_services[service_type]:
+            return  # already the default
```

Comment:
But it might be the global default and user is trying to set it as the project-specific default (so that it won't change for this project if the global default is later changed)